R/FilterState.R

```diff
@@ -304,10 +305,15 @@ FilterState <- R6::R6Class( # nolint
           private$dataname
         )
       )
-      value <- private$cast_and_validate(value)
-      value <- private$remove_out_of_bound_values(value)
-      private$validate_selection(value)
-      private$selected(value)
+      if (private$is_disabled()) {
+        private$selected(value)
```

Comment:
I think we should throw warning if it's disabled and avoid setting value. Setting value of disabled filter is confusing mechanism which causes `$get_selected` to return values and `$get_call` to return condition. Enabling will overwrite these values anyway, so they are redundant and potentially could cause some bugs.

Reply:
I disagree. I think it's misleading if the filter state is disabled but there are  values set, which is the case if we don't set the state to `NULL`. If you run one of the apps in #165 on this branch, you'll see the main output shows that a disabled filter state has values. To me that implies the filter is active.


As far as `$get_selected` and `$get_call`, if the filter state is disabled, and the state set to `NULL`, these methods will return `NULL` so they won't apply any filters to the data. The [adjustment to `calls_combine_by`](https://github.com/insightsengineering/teal.slice/blob/990940a2d621e6b486e09c460014adbd57cfd77d/R/call_utils.R#L468) for example ignores any calls for a disabled filter.